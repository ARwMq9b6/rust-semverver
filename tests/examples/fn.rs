pub mod old {
    pub fn abc() {}

    pub fn bcd() {}

    pub fn cde() {}

    pub fn def(_: u8) {}

    pub fn efg(a: u8, _: u8) -> u8 {
        a
    }

    pub fn fgh(a: u8, _: u8) -> u8 {
        a
    }

    pub fn ghi(a: u8, _: u8) -> u8 {
        a
    }
}

pub mod new {
    pub fn abc() {}

    pub fn bcd(_: u8) {}

    pub fn cde() -> u16 {
        0xcde
    }

    pub fn def() {}

    // TODO: this should be non-breaking (even after we make TypeParameterAdded something
    // that is not `Breaking`)
    pub fn efg<A>(a: A, _: A) -> A {
        a
    }

    pub fn fgh(a: u8, _: u16) -> u8 {
        a
    }

    pub fn ghi(a: u8, _: u8) -> u16 {
        a as u16
    }
}
