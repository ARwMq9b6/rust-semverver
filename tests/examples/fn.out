version bump: 1.0.0 -> (Breaking) -> 2.0.0
warning: Breaking changes in `bcd`
  --> $REPO_PATH/tests/examples/fn.rs:26:5
   |
26 |     pub fn bcd(_: u8) {}
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = warning: Breaking (TypeChanged { error: ArgCount })

warning: Breaking changes in `cde`
  --> $REPO_PATH/tests/examples/fn.rs:28:5
   |
28 | /     pub fn cde() -> u16 {
29 | |         0xcde
30 | |     }
   | |_____^
   |
   = warning: Breaking (TypeChanged { error: Sorts(ExpectedFound { expected: (), found: u16 }) })

warning: Breaking changes in `def`
  --> $REPO_PATH/tests/examples/fn.rs:32:5
   |
32 |     pub fn def() {}
   |     ^^^^^^^^^^^^^^^
   |
   = warning: Breaking (TypeChanged { error: ArgCount })

warning: NonBreaking changes in `efg`
  --> $REPO_PATH/tests/examples/fn.rs:36:5
   |
36 | /     pub fn efg<A>(a: A, _: A) -> A {
37 | |         a
38 | |     }
   | |_____^
   |
   = note: NonBreaking (TypeParameterAdded { defaulted: true })

warning: Breaking changes in `fgh`
  --> $REPO_PATH/tests/examples/fn.rs:40:5
   |
40 | /     pub fn fgh(a: u8, _: u16) -> u8 {
41 | |         a
42 | |     }
   | |_____^
   |
   = warning: Breaking (TypeChanged { error: Sorts(ExpectedFound { expected: u8, found: u16 }) })

warning: Breaking changes in `ghi`
  --> $REPO_PATH/tests/examples/fn.rs:44:5
   |
44 | /     pub fn ghi(a: u8, _: u8) -> u16 {
45 | |         a as u16
46 | |     }
   | |_____^
   |
   = warning: Breaking (TypeChanged { error: Sorts(ExpectedFound { expected: u8, found: u16 }) })

